### PR TITLE
hv: coding style: refine functions under dm/ to one exit point

### PR DESCRIPTION
**Pull request by lifeix**
Refine vioapic_indirect_write and vmsix_table_rw to one exit point.

Tracked-On: #861
Signed-off-by: Li, Fei1 <fei1.li@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2256<br>No new violations to the coding guideline detected.

File: hypervisor/dm/vioapic.c [hunk 1]
```diff
@@ -304,8 +302,8 @@ vioapic_indirect_write(struct acrn_vioapic *vioapic, uint32_t addr,
 	}
 
 	/* redirection table entries */
-	if ((regnum >= IOAPIC_REDTBL) &&
-	    (regnum < (IOAPIC_REDTBL + (pincount * 2U)))) {
+	if ((regnum >= IOAPIC_REDTBL) && (regnum < (IOAPIC_REDTBL + (pincount * 2U)))) {
+		bool status = true;
```
Comment (dongyaozu) [inline hypervisor/dm/vioapic.c]: Don't call it status. Status = true/false didn't tell us what it is for.

File: hypervisor/dm/vioapic.c [hunk 2]
```diff
@@ -334,73 +332,63 @@ vioapic_indirect_write(struct acrn_vioapic *vioapic, uint32_t addr,
...
-			uint16_t i;
-			struct acrn_vcpu *vcpu;
 
-			dev_dbg(ACRN_DBG_IOAPIC,
-			"ioapic pin%hhu: recalculate vlapic trigger-mode reg",
-			pin);
-
-			foreach_vcpu(i, vioapic->vm, vcpu) {
-				vcpu_make_request(vcpu, ACRN_REQUEST_TMR_UPDATE);
+		if (status == true) {
```
Comment (dongyaozu) [inline hypervisor/dm/vioapic.c]: Bool. Use it  directly.

Comment (lifeix) [reply hypervisor/dm/vioapic.c]: done

File: hypervisor/dm/vioapic.c (hunk 1 above)
Comment (lifeix) [reply hypervisor/dm/vioapic.c]: rename it to wire_mode_valid

**Comment by HuiHuangShi** [comment]
2256<br>No new violations to the coding guideline detected.

**Pull request merged**